Scripts/UtilityScripts/README.md

```diff
@@ -0,0 +1,239 @@
+<br />Purpose: Adds a network access rule to the Key Vault for each provided public IP address. Also sets public network access enabled with default action Deny.
+<br />Parameters: SubscriptionId, ResourceGroupName, KeyVaultName, PublicIpAddresses
+
+`Set-KeyVaultPublicNetworkAccessEnabledForIpAddress()`
+<br />Purpose: Adds a network access rule to the Key Vault for the provided public IP address. Also sets public network access enabled with default action Deny.
+<br />Parameters: SubscriptionId, ResourceGroupName, KeyVaultName, PublicIpAddress
+
+`Remove-KeyVaultNetworkAccessRuleForIpAddress()`
+<br />Purpose: Removes a network access rule for the provided public IP address from the Key Vault.
+<br />Parameters: SubscriptionId, ResourceGroupName, KeyVaultName, PublicIpAddress
+
+<br />
+
+### Azure_Subscription_AuthZ_Remove_Deprecated_Accounts
+
+`Get-DeletedUser()`
+<br />Purpose: Retrieves the recently deleted object from Microsoft Graph's deleted objects. Useful when trying to match a name to a deleted object ID (e.g. user or service principal GUID).
+<br />Parameters: SubscriptionId, DeletedObjectId
+
+<br />
+
+### Azure_Subscription_DP_Avoid_Plaintext_Secrets_Deployments
+
+`Get-ResourceGroupDeployment()`
+<br />Purpose: Show the specified Azure Resource Group deployment (or list all if no deployment is specified by name). Useful to check for secrets on Deployment parameters.
+<br />Parameters: SubscriptionId, ResourceGroupName, DeploymentName
+
+`Get-ResourceGroupDeploymentOperations()`
+<br />Purpose: Show the Azure Resource Group deployment's detailed operations.
+<br />Parameters: SubscriptionId, ResourceGroupName, DeploymentName
+
+`Get-SubscriptionDeployments()`
+<br />Purpose: Show Azure Subscription deployments. Useful to check for secrets on Deployment parameters.
+<br />Parameters: SubscriptionId
+
+`Get-SubscriptionDeploymentsAndOperations()`
+<br />Purpose: Show Azure Subscription deployments as well as detailed operations for each. Useful to check for secrets on Deployment parameters.
+<br />Parameters: SubscriptionId
+
+<br />
+
+### Azure_Subscription_SI_Dont_Use_B2C_Tenant
+
+`Get-AzureADB2CTenants()`
+<br />Purpose: List all Azure AD B2C tenants deployed in the subscription.
+<br />Parameters: SubscriptionId
+
+`Get-AzureADB2CResourceProvider()`
+<br />Purpose: Show the Azure AD B2C Resource Provider and its current registration state on the subscription.
+<br />Parameters: SubscriptionId
+
+`Get-RegisteredResourceProviders()`
+<br />Purpose: List all Azure Resource Providers registered in the subscription.
+<br />Parameters: SubscriptionId
+
+`Unregister-AzureADB2CResoureProvider()`
+<br />Purpose: Unregister the Azure AD B2C Resource Provider from the subscription.
+<br />Parameters: SubscriptionId
+
+<br />
+
+### Azure_VirtualMachine_SI_Enable_Antimalware
+
+`Get-MDEPreferences()`
+<br />Purpose: Run on a VM to show current configuration preferences for Microsoft Defender for Endpoint.
```

Comment:
NIT: 'Run on a VM' just this might not be able to convey required info that these scripts has to be run form inside the VM. Just a suggestion.